File: move-index.js
const fs = require('fs');
const path = require('path');

// Create public directory if it doesn't exist
const publicDir = path.join(__dirname, 'public');
if (!fs.existsSync(publicDir)) {
    fs.mkdirSync(publicDir, { recursive: true });
    console.log('Created public directory');
}

// Read index.html from current directory
const indexPath = path.join(__dirname, 'index.html');
const publicIndexPath = path.join(publicDir, 'index.html');

if (fs.existsSync(indexPath)) {
    const content = fs.readFileSync(indexPath, 'utf8');
    fs.writeFileSync(publicIndexPath, content);
    console.log('Copied index.html to public directory');
    
    // Delete the original file
    fs.unlinkSync(indexPath);
    console.log('Deleted original index.html');
    
    // Verify
    if (fs.existsSync(publicIndexPath)) {
        console.log('Verification successful: public/index.html exists');
    }
} else {
    console.log('index.html not found in current directory');
}

File: move_index.py
import os
import shutil
from pathlib import Path

# Define paths
project_dir = Path("C:\\Users\\hoffmannm23\\Desktop\\Projects\\SpaceyMiner")
public_dir = project_dir / "public"
index_file = project_dir / "index.html"
destination_file = public_dir / "index.html"

# Create public directory if it doesn't exist
if not public_dir.exists():
    public_dir.mkdir(parents=True, exist_ok=True)
    print("Created public directory")
else:
    print("Public directory already exists")

# Move index.html to public directory
if index_file.exists():
    shutil.move(str(index_file), str(destination_file))
    print("Moved index.html to public directory")
else:
    print("index.html not found in current directory")

# Verify the file is in the public directory
if destination_file.exists():
    file_size = destination_file.stat().st_size
    print(f"✓ Verification successful: public/index.html exists ({file_size} bytes)")
    print(f"  Location: {destination_file}")
else:
    print("✗ Verification failed: public/index.html not found")
